cli/src/semgrep/rule_match.py

```diff
@@ -365,6 +366,19 @@ def get_match_based_key(self) -> Tuple:
                 path,
                 self.annotated_rule_name,
             )
+
+        # Check if the match is non-code-match and reachable is False
+        if "sca_info" in self.extra and not self.extra["sca_info"].reachable:
+            # Include dependency package and version in the key
```

Comment:
How does this deal with two copies of the same package at the same version in a single lockfile?